plugins/modules/aws_s3_bucket_info.py

```diff
@@ -57,50 +182,223 @@
+    for bucket in buckets:
+        bucket.update(get_bucket_details(connection, bucket['name'], requested_facts, transform_location))
+        full_bucket_list.append(bucket)
+
+    return(full_bucket_list)
+
+
+def get_bucket_details(connection, name, requested_facts, transform_location):
+    """
+    Execute all enabled S3API get calls for selected bucket
+    """
+    all_facts = {}
+
+    for key in requested_facts:
+        if requested_facts[key]:
+            if key == 'bucket_location':
+                all_facts[key] = {}
+                try:
+                    all_facts[key] = get_bucket_location(name, connection, transform_location)
+                # we just pass on error - error means that resources is undefined
+                except botocore.exceptions.ClientError:
+                    pass
+            elif key == 'bucket_tagging':
+                all_facts[key] = {}
+                try:
+                    all_facts[key] = get_bucket_tagging(name, connection)
+                # we just pass on error - error means that resources is undefined
+                except botocore.exceptions.ClientError:
+                    pass
+            else:
+                all_facts[key] = {}
+                try:
+                    all_facts[key] = get_bucket_property(name, connection, key)
+                # we just pass on error - error means that resources is undefined
+                except botocore.exceptions.ClientError:
+                    pass
+
+    return(all_facts)
+
+
+@AWSRetry.exponential_backoff(max_delay=120, catch_extra_error_codes=['NoSuchBucket', 'OperationAborted'])
+def get_bucket_location(name, connection, transform_location=False):
+    """
+    Get bucket location and optionally transform 'null' to 'us-east-1'
+    """
+    data = connection.get_bucket_location(Bucket=name)
+
+    # Replace 'null' with 'us-east-1'?
+    if transform_location:
+        try:
+            if not data['LocationConstraint']:
+                data['LocationConstraint'] = 'us-east-1'
+        except KeyError:
+            pass
+    # Strip response metadata (not needed)
+    try:
+        data.pop('ResponseMetadata')
+        return(data)
+    except KeyError:
+        return(data)
+
+
+@AWSRetry.exponential_backoff(max_delay=120, catch_extra_error_codes=['NoSuchBucket', 'OperationAborted'])
+def get_bucket_tagging(name, connection):
+    """
+    Get bucket tags and transform them using `boto3_tag_list_to_ansible_dict` function
+    """
+    data = connection.get_bucket_tagging(Bucket=name)
+
+    try:
+        bucket_tags = boto3_tag_list_to_ansible_dict(data['TagSet'])
+        return(bucket_tags)
+    except KeyError:
+        # Strip response metadata (not needed)
+        try:
+            data.pop('ResponseMetadata')
+            return(data)
+        except KeyError:
+            return(data)
+
+
+@AWSRetry.exponential_backoff(max_delay=120, catch_extra_error_codes=['NoSuchBucket', 'OperationAborted'])
```

Comment:
I'd generally recommend using jittered_backoff instead of exponential (It's basically an exponential backoff with some randomness added: https://aws.amazon.com/blogs/architecture/exponential-backoff-and-jitter/